박수진/week1/app/src/main/java/com/example/week1/data/network/NetworkState.kt

```diff
@@ -0,0 +1,15 @@
+package com.example.week1.data.network
+
+enum class Status {
+    RUNNING,
+    SUCCESS,
+    FAILED
+}
+
+class NetworkState(val status: Status, val msg: String) {
+    companion object {
+        val LOADED: NetworkState = NetworkState(Status.SUCCESS, "Success")
+        val LOADING: NetworkState = NetworkState(Status.RUNNING, "Running")
+        val ERROR: NetworkState = NetworkState(Status.FAILED, "Something went wrong")
+    }
+}
```

Comment:
만약 위처럼 status, msg가 고정으로 사용하고 싶다면 enum을 그대로 사용해도 좋을 듯해요
```suggestion


enum class NetworkState(val status: Status, val msg: String) {
     LOADED(Status.SUCCESS, "Success"),
     LOADING(Status.RUNNING, "Running")
     ERROR(Status.FAILED, "Something went wrong")
}
```

Reply:
아하 바로 enum class로 작성해도 괜찮겠군용 말씀하신걸 참고해서 밑의 코드로 수정했습니다 SUCCESS, RUNNING, FAILED가 LOADED, LOADING, ERROR와 중복되는 느낌이고 단순히 상태만 확인하려 선언한 클래스라 msg만 남겨두고 Status는 삭제했어요 감사합니다(o゜▽゜)o☆

```
enum class NetworkState(val msg: String) {
    LOADED("Success"),
    LOADING("Running"),
    ERROR("Something went wrong")
}
```

Reply:
짱짱!!